lib/elixir/lib/uri.ex

```diff
@@ -440,12 +444,20 @@ defmodule URI do
 
   # Regex.run returns empty strings sometimes. We want
   # to replace those with nil for consistency.
-  defp nillify(list) do
+  defp nillify(list) when is_list(list) do
```

Comment:
Do we still need this version? If something else is using it, should rewrite it to also be explicit like above?